app/lib/actions/katello/content_view_version/incremental_update.rb

```diff
@@ -83,6 +87,30 @@ def components_repo_instance(repo, component_versions)
           possible_repos.flatten.first
         end
 
+        def run
+          content = { ::Katello::Erratum::CONTENT_TYPE => [],
+                      ::Katello::Package::CONTENT_TYPE => [],
+                      ::Katello::PuppetModule::CONTENT_TYPE => []}
+
+          input[:copy_action_outputs].each do |output|
+            output[:pulp_tasks].each do |pulp_task|
+              pulp_task[:result][:units_successful].each do |unit|
+                type = unit['type_id']
+                unit = unit['unit_key']
+                case type
+                when ::Katello::Erratum::CONTENT_TYPE
+                  content[::Katello::Erratum::CONTENT_TYPE] <<  unit['id']
+                when ::Katello::Package::CONTENT_TYPE
+                  content[::Katello::Package::CONTENT_TYPE] <<  ::Katello::Util::Package.build_nvra(unit)
+                when ::Katello::PuppetModule::CONTENT_TYPE
+                  content[::Katello::PuppetModule::CONTENT_TYPE] <<  "#{unit['author']}-#{unit['name']}-#{unit['version']}"
+                end
+              end
+            end
+          end
+          output[:added_units] = content
```

Comment:
I find this kind of confusing since you are using `output` as the argument to your each block in line 95.